### PR TITLE
8329767: G1: Move G1BlockOffsetTable::set_for_starts_humongous to HeapRegion

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

This patch moves the method `G1BlockOffsetTable::set_for_starts_humongous` into `HeapRegion::set_starts_humongous` and marks `G1BlockOffsetTable::update_for_block(HeapWord* blk_start, size_t size)` as public (because it delegates to another public method which has the same name).

Thanks for taking the time to review.

Best Regards,
-- Guoxiong

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8329767](https://bugs.openjdk.org/browse/JDK-8329767): G1: Move G1BlockOffsetTable::set_for_starts_humongous to HeapRegion (**Enhancement** - P5)

### Reviewers
 * [Albert Mingkun Yang](https://openjdk.org/census#ayang) (@albertnetymk - **Reviewer**)
 * [Ivan Walulya](https://openjdk.org/census#iwalulya) (@walulyai - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18701/head:pull/18701` \
`$ git checkout pull/18701`

Update a local copy of the PR: \
`$ git checkout pull/18701` \
`$ git pull https://git.openjdk.org/jdk.git pull/18701/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18701`

View PR using the GUI difftool: \
`$ git pr show -t 18701`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18701.diff">https://git.openjdk.org/jdk/pull/18701.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18701#issuecomment-2045654819)

### CODE REVIEW

File: src/hotspot/share/gc/g1/g1BlockOffsetTable.hpp [hunk 1]
```diff
@@ -128,7 +124,9 @@ class G1BlockOffsetTable: public CHeapObj<mtGC> {
     }
   }
 
-  void set_for_starts_humongous(HeapRegion* hr, HeapWord* obj_top, size_t fill_size);
+  void update_for_block(HeapWord* blk_start, size_t size) {
```
Comment (albertnetymk) [inline src/hotspot/share/gc/g1/g1BlockOffsetTable.hpp:127]: Maybe this can be done at the caller side? (The API would be easier with fewer overloading.)

Comment (lgxbslgx) [reply src/hotspot/share/gc/g1/g1BlockOffsetTable.hpp]: I removed the method `void update_for_block(HeapWord* blk_start, size_t size)` just now.

**Review by albertnetymk — APPROVED** [review]

**Review by walulyai — APPROVED** [review]
LGTM!

**Comment by lgxbslgx** [comment]
@albertnetymk @walulyai Thanks for your reviews.

/integrate

**Pull request closed without merging**